probe/kubernetes/reporter.go

```diff
@@ -658,26 +654,13 @@ func (r *Reporter) podTopology(services []Service, deployments []Deployment, dae
 			))
 		}
 	}
-
-	var localPodUIDs map[string]struct{}
-	if r.nodeName == "" && r.kubeletPort != 0 {
-		// We don't know the node name: fall back to obtaining the local pods from kubelet
-		var err error
-		localPodUIDs, err = GetLocalPodUIDs(fmt.Sprintf("127.0.0.1:%d", r.kubeletPort))
-		if err != nil {
-			log.Warnf("No node name and cannot obtain local pods, reporting all (which may impact performance): %v", err)
-		}
+	// filter out non-local pods: we only want to report local ones for performance reasons.
+	if r.nodeName == "" {
```

Comment:
I think, on rereading this, that the nodeName=="" check should be inside rather than outside the WalkPods function call. right now this is not allowing a cluster agent to get the node for any pod. If inside it will move on to the pods.AddNode() call which will allow it to attach a node to a pod.

Reply:
Changed the order of this check in latest commit.